src/Compilers/Core/Portable/StrongName/PortableStrongNameProvider.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Cci;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class PortableStrongNameProvider : StrongNameProvider
+    {
+        private readonly ImmutableArray<string> _keyFileSearchPaths;
+        internal StrongNameFileSystem FileSystem { get; set; }
+
+        public PortableStrongNameProvider(ImmutableArray<string> keySearchPaths, StrongNameFileSystem strongNameFileSystem)
+        {
+            FileSystem = strongNameFileSystem ?? StrongNameFileSystem.s_StrongNameFileSystemInstance;
+            _keyFileSearchPaths = keySearchPaths.NullToEmpty();
+        }
+
+        public override int GetHashCode()
+        {
+            return 0;
+        }
+
+        internal override SigningCapability Capability => SigningCapability.SignsPeBuilder;
+
+        internal override StrongNameKeys CreateKeys(string keyFilePath, string keyContainerName, CommonMessageProvider messageProvider)
+        {
+            var keyPair = default(ImmutableArray<byte>);
+            var publicKey = default(ImmutableArray<byte>);
+            string container = null;
+
+            if (!string.IsNullOrEmpty(keyFilePath))
```

Comment:
Is `keyFilePath` ever null or empty?

Reply:
Yeah, keyFilePath can be null if kyeContainerName is not null.  Portable StrongNameProvider doesn't support key containers.